src/CommunityToolkit.Aspire.Hosting.Flagd/FlagdBuilderExtensions.cs

```diff
@@ -0,0 +1,56 @@
+using Aspire.Hosting.ApplicationModel;
+using CommunityToolkit.Aspire.Hosting.Flagd;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding flagd resources to an <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class FlagdBuilderExtensions
+{
+    /// <summary>
+    /// Adds a flagd container to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="fileSource">The path to the flag configuration file on the host. The flags configuration should be stored in a file named flagd.json</param>
+    /// <param name="port">The host port for flagd HTTP endpoint. If not provided, a random port will be assigned.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{FlagdResource}"/>.</returns>
+    public static IResourceBuilder<FlagdResource> AddFlagd(
+        this IDistributedApplicationBuilder builder,
+        [ResourceName] string name,
+        string fileSource,
+        int port = 8013)
```

Comment:
todo: the other integration uses `int? port = null` so that it provides a dynamic port by default, allowing the possibility to override the port only when needed.